### PR TITLE
[3.x] Fix ViewportTexture error when viewport is used from a child scene

### LINKED ISSUE #27790: ViewportTexture: Path to node is invalid
**Godot version:** v3.1.stable.mono.official

**OS/device including version:** Windows 10.0.17763 Build 17763

**Issue description:**

I created a Viewport node and had it draw some GUI. Then, on a MeshInstance (quad) I set its albedo texture to ViewportTexture and selected the viewport node (at Player/Viewport). It still came up with a preview.

When I ran the game I was met with two errors:
```
Node not found: Player/Viewport
ViewportTexture: Path to node is invalid
```
But I was still able to play the game and furthermore, the viewport still worked correctly. I took a look at the scene's `.tscn` file and found that the ViewportTexture does indeed point correctly to the Viewport node I had created.

ViewportTexture, SpatialMaterial, and QuadMesh:
```tscn
...

[sub_resource type="ViewportTexture" id=1]
flags = 1
viewport_path = NodePath("Player/Viewport")

[sub_resource type="SpatialMaterial" id=2]
resource_local_to_scene = true
flags_transparent = true
flags_unshaded = true
flags_albedo_tex_force_srgb = true
albedo_texture = SubResource( 1 )

[sub_resource type="QuadMesh" id=3]
resource_local_to_scene = true
material = SubResource( 2 )
size = Vector2( 0.3, 0.2 )

...
```

**Scene tree:**
```
 ┖╴Spatial
    ┠╴Player
    ┃  ┠╴Camera
    ┃  ┠╴Cockpit
    ┃  ┠╴Right
    ┃  ┠╴Left
    ┃  ┠╴CockpitAudio
    ┃  ┠╴GunFireCockpitAudio
    ┃  ┠╴TargetingMesh
    ┃  ┖╴Viewport
    ┃     ┖╴GUI
    ┃        ┖╴Speed
    ┠╴CSGBox
    ┖╴CSGBox2
```

This is very similar to #14790, #12781, and #19259.
**Steps to reproduce:**
Create a Viewport node, assign a ViewportTexture to a mesh, select the Viewport you made, and run the game.

### PR DESCRIPTION
**Pull request by jitspoe**
This should hopefully fix some of the viewport errors mentioned in https://github.com/godotengine/godot/issues/27790

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -53,30 +53,32 @@
 #include "servers/physics_2d_server.h"
 
 void ViewportTexture::setup_local_to_scene() {
-	if (vp) {
-		vp->viewport_textures.erase(this);
-	}
+	if (get_local_scene()) {
```
Comment (akien-mga) [inline scene/main/viewport.cpp:56]: We'd save on indentation and Git diff with:
```
if (!get_local_scene()) {
	return;
}
```

**Comment by akien-mga** [comment]
Is this also relevant for 4.0?

**Comment by jitspoe** [comment]
> Is this also relevant for 4.0?

That's a good question.  I think when I initially made the change, I was having trouble getting 4.0 to run.  I seem to remember something about the viewport stuff changing, but looking at master, that function appears to be the same.  Perhaps I should do the proper fix in master then pull it over to 3.x.

**Comment by akien-mga** [comment]
I pushed an update to improve the code changes, as the check was already being done just a couple lines below.

**Comment by jitspoe** [comment]
Thanks for taking care of that.  Was off on vacation + busy with other stuff, so I didn't get around to cleaning this up.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**